tests/dir_helpers.py

```diff
@@ -253,12 +253,33 @@ class PosixTmpDir(TmpDir, pathlib.PurePosixPath):
     pass
 
 
+CACHE = {}
+
+
 @pytest.fixture(scope="session")
-def make_tmp_dir(tmp_path_factory, request):
-    def make(name, *, scm=False, dvc=False, **kwargs):
+def make_tmp_dir(tmp_path_factory, request, worker_id):
+    def make(name, *, scm=False, dvc=False, subdir=False):
+        from dvc.repo import Repo
+        from dvc.scm.git import Git
+        from dvc.utils.fs import fs_copy
+
+        cache = CACHE.get((scm, dvc, subdir))
+        if not cache:
+            cache = tmp_path_factory.mktemp("dvc-test-cache" + worker_id)
+            TmpDir(cache).init(scm=scm, dvc=dvc, subdir=subdir)
+            CACHE[(scm, dvc, subdir)] = os.fspath(cache)
```

Comment:
Could've just `= cache` and use Path down below instead of listdir/os.path :facepalm: